JSQMessagesViewController/Model/JSQVideoMediaItem.m

```diff
@@ -79,20 +97,51 @@ - (UIView *)mediaView
-        
-        UIImageView *imageView = [[UIImageView alloc] initWithImage:playIcon];
-        imageView.backgroundColor = [UIColor blackColor];
-        imageView.frame = CGRectMake(0.0f, 0.0f, size.width, size.height);
-        imageView.contentMode = UIViewContentModeCenter;
-        imageView.clipsToBounds = YES;
-        [JSQMessagesMediaViewBubbleImageMasker applyBubbleImageMaskToMediaView:imageView isOutgoing:self.appliesMediaViewMaskAsOutgoing];
-        self.cachedVideoImageView = imageView;
+
+    if (self.cachedVideoImageView == nil)
+    {
+        if (_image == nil)
+        {
+            CGSize size = [self mediaViewDisplaySize];
+            UIImage *playIcon = [[UIImage jsq_defaultPlayImage] jsq_imageMaskedWithColor:[UIColor lightGrayColor]];
+
+            UIImageView *imageView = [[UIImageView alloc] initWithImage:playIcon];
+            imageView.backgroundColor = [UIColor blackColor];
+            imageView.frame = CGRectMake(0.0f, 0.0f, size.width, size.height);
+            imageView.contentMode = UIViewContentModeCenter;
+            imageView.clipsToBounds = YES;
+            [JSQMessagesMediaViewBubbleImageMasker applyBubbleImageMaskToMediaView:imageView isOutgoing:self.appliesMediaViewMaskAsOutgoing];
+            self.cachedVideoImageView = imageView;
+        }
+        else
+        {
+            CGSize size = [self mediaViewDisplaySize];
+            UIImage *playIcon = [[UIImage jsq_defaultPlayImage] jsq_imageMaskedWithColor:[UIColor lightGrayColor]];
+
+            CGRect scaledImageRect = CGRectZero;
+            CGFloat aspectWidth = _image.size.width / playIcon.size.width;
+            CGFloat aspectHeight = _image.size.height / playIcon.size.height;
+            CGFloat aspectRatio = MIN ( aspectWidth, aspectHeight );
+            scaledImageRect.size.width = playIcon.size.width * aspectRatio / 5;
+            scaledImageRect.size.height = playIcon.size.height * aspectRatio / 5;
+            scaledImageRect.origin.x = (_image.size.width - scaledImageRect.size.width) / 2.0f;
+            scaledImageRect.origin.y = (_image.size.height - scaledImageRect.size.height) / 2.0f;
+
+            UIGraphicsBeginImageContextWithOptions(_image.size, NO, 0.0f);
+            [_image drawInRect:CGRectMake(0, 0, _image.size.width, _image.size.height)];
+            [playIcon drawInRect:scaledImageRect];
+            UIImage *resultImage = UIGraphicsGetImageFromCurrentImageContext();
+            UIGraphicsEndImageContext();
+
+            UIImageView *imageView = [[UIImageView alloc] initWithImage:resultImage];
+            imageView.frame = CGRectMake(0.0f, 0.0f, size.width, size.height);
+            imageView.contentMode = UIViewContentModeScaleAspectFill;
+            imageView.clipsToBounds = YES;
+            [JSQMessagesMediaViewBubbleImageMasker applyBubbleImageMaskToMediaView:imageView isOutgoing:self.appliesMediaViewMaskAsOutgoing];
+            self.cachedVideoImageView = imageView;
```

Comment:
another helper method for the `else` block, too:

`configureVideoViewWithCustomThumbnail`